### PR TITLE
Refactor and rename

### LINKED ISSUE #19: Renaming effects - alignment
_From @Depechie on February 16, 2017 7:15_

We are looking to rename the current effects, to streamline the naming.
But we are a bit stuck with the current proposal!

Proposal:
Current name : public class CapitalizeKeyboardEffect : RoutingEffect
New name : public class EntryCapitalizeKeyboard : RoutingEffect
So prefix the control it targets and loose the suffix Effect

But if we uphold this, we get in trouble with the static classes that are needed for adding dependency properties to pass through to the actual RoutingEffect.

Problem:
public static class ChangeColorEffect with tandem class > public class ChangeColorSwitchEffect : RoutingEffect ( https://github.com/FormsCommunityToolkit/Effects/blob/dev/src/Effects/ChangeColorEffect.cs )
public static class ChangePickerColorEffect with tandem class > public class ChangeColorPickerEffect : RoutingEffect ( https://github.com/FormsCommunityToolkit/Effects/blob/dev/src/Effects/ChangePickerColorEffect.cs )

How do we rename these? ChangeColorEffect would become SwitchChangeColor, but what would ChangeColorSwitchEffect be then?
Same with the picker class...

Suggestions?

_Copied from original issue: FormsCommunityToolkit/Effects#48_

### PR DESCRIPTION
**Pull request by Depechie**
Fixes #19

Changes Proposed in this pull request:
Trying to align all effects in reference to naming

### CODE REVIEW

**Comment by bartlannoeye** [comment]
@jamesmontemagno any reason for AppVeyor "fail"? I don't see why.

File: src/Effects/Effects.Android/Effects/Entry/EntryCapitalizeKeyboard.cs [hunk 1]
```diff
@@ -6,28 +6,28 @@
 using Android.Runtime;
 using System.Linq;
 
-[assembly: ExportEffect(typeof(CapitalizeKeyboardEffect), nameof(CapitalizeKeyboardEffect))]
+[assembly: ExportEffect(typeof(EntryCapitalizeKeyboard), nameof(EntryCapitalizeKeyboard))]
```
Comment (bartlannoeye) [inline src/Effects/Effects.Android/Effects/Entry/EntryCapitalizeKeyboard.cs]: Do we want a newline here? I'd remove it and put attribute right on top of namespace.

File: src/Effects/Effects.Android/Effects/Entry/EntryClear.cs [hunk 2]
```diff
@@ -5,11 +5,11 @@
 using Xamarin.Forms;
 using Xamarin.Forms.Platform.Android;
 
-[assembly: ExportEffect(typeof(ClearEntryEffect), nameof(ClearEntryEffect))]
+[assembly: ExportEffect(typeof(EntryClear), nameof(EntryClear))]
 namespace FormsCommunityToolkit.Effects.Droid
 {
     [Preserve(AllMembers = true)]
-    public class ClearEntryEffect : PlatformEffect
+    public class EntryClear : PlatformEffect
```
Comment (bartlannoeye) [inline src/Effects/Effects.Android/Effects/Entry/EntryClear.cs]: Should we remove empty overrides? Or leave them as placeholder for future extension (YAGNI ?)

File: src/Effects/Effects.Android/Effects/Label/LabelCustomFont.cs [hunk 3]
```diff
@@ -18,7 +18,7 @@ protected override void OnAttached()
                 return;
             else
             {
-                var effect = (FormsCommunityToolkit.Effects.CustomFontEffect)Element.Effects.FirstOrDefault(item => item is FormsCommunityToolkit.Effects.CustomFontEffect);
+                var effect = (FormsCommunityToolkit.Effects.LabelCustomFont)Element.Effects.FirstOrDefault(item => item is FormsCommunityToolkit.Effects.LabelCustomFont);
```
Comment (bartlannoeye) [inline src/Effects/Effects.Android/Effects/Label/LabelCustomFont.cs]: simplify typename (drop namespace part)

File: src/Effects/Effects.Android/Effects/Picker/PickerChangeColor.cs [hunk 4]
```diff
@@ -1,25 +1,24 @@
...
 using Xamarin.Forms;
 using Xamarin.Forms.Platform.Android;
 using Color = Xamarin.Forms.Color;
 using Picker = Android.Widget.EditText;
+using PickerChangeColor = FormsCommunityToolkit.Effects.Droid.PickerChangeColor;
 
-[assembly: ExportEffect(typeof(ChangeColorPickerEffect), nameof(ChangeColorPickerEffect))]
+[assembly: ExportEffect(typeof(PickerChangeColor), nameof(PickerChangeColorEffect))]
 namespace FormsCommunityToolkit.Effects.Droid
 {
```
Comment (bartlannoeye) [inline src/Effects/Effects.Android/Effects/Picker/PickerChangeColor.cs]: remove newline

File: src/Effects/Effects.Android/Effects/Picker/PickerChangeColor.cs [hunk 5]
```diff
@@ -1,25 +1,24 @@
...
-    public class ChangeColorPickerEffect : PlatformEffect
+    public class PickerChangeColor : PlatformEffect
     {
         private Color _color;
 
         protected override void OnAttached ()
         {
-            
-            _color = (Color)Element.GetValue(ChangePickerColorEffect.ColorProperty);
+            _color = (Color)Element.GetValue(FormsCommunityToolkit.Effects.PickerChangeColor.ColorProperty);
```
Comment (bartlannoeye) [inline src/Effects/Effects.Android/Effects/Picker/PickerChangeColor.cs]: Simplify typename

File: src/Effects/Effects.UWP/Effects/Entry/EntryRemoveBorder.cs [hunk 6]
```diff
@@ -2,33 +2,31 @@
...
-[assembly: ExportEffect(typeof(RemoveBorderEffect), nameof(RemoveBorderEffect))]
 
+[assembly: ExportEffect(typeof(EntryRemoveBorder), nameof(EntryRemoveBorder))]
 namespace FormsCommunityToolkit.Effects.UWP
 {
-    public class RemoveBorderEffect : PlatformEffect
+    public class EntryRemoveBorder : PlatformEffect
     {
-        Thickness old;
+        private Windows.UI.Xaml.Thickness _old;
```
Comment (bartlannoeye) [inline src/Effects/Effects.UWP/Effects/Entry/EntryRemoveBorder.cs]: typename

File: src/Effects/Effects.UWP/Effects/Label/LabelMultiLine.cs [hunk 7]
```diff
@@ -21,10 +21,10 @@ protected override void OnAttached()
                 return;
             else
             {
-                initialeLines = control.MaxLines;
-                initialTextWrapping = control.TextWrapping;
+                _initialeLines = control.MaxLines;
+                _initialTextWrapping = control.TextWrapping;
 
-                var effect = (FormsCommunityToolkit.Effects.MultiLineLabelEffect)Element.Effects.FirstOrDefault(item => item is FormsCommunityToolkit.Effects.MultiLineLabelEffect);
+                var effect = (FormsCommunityToolkit.Effects.LabelMultiLine)Element.Effects.FirstOrDefault(item => item is FormsCommunityToolkit.Effects.LabelMultiLine);
```
Comment (bartlannoeye) [inline src/Effects/Effects.UWP/Effects/Label/LabelMultiLine.cs]: typenames

File: src/Effects/Effects.UWP/Effects/Picker/PickerChangeColor.cs [hunk 8]
```diff
@@ -1,22 +1,22 @@
 ﻿using Windows.UI.Xaml.Controls;
-using FormsCommunityToolkit.Effects.UWP.Effects;
 using Xamarin.Forms;
 using Xamarin.Forms.Internals;
 using Xamarin.Forms.Platform.UWP;
 using Windows.UI.Xaml.Media;
+using FormsCommunityToolkit.Effects;
+using PickerChangeColor = FormsCommunityToolkit.Effects.UWP.Effects.PickerChangeColor;
```
Comment (bartlannoeye) [inline src/Effects/Effects.UWP/Effects/Picker/PickerChangeColor.cs]: these seem to give issues :)

What about
using RoutingEffects = FormCommunityToolkit.Effects;
using PickerChangeColor = FormsCommunityToolkit.Effects.UWP.Effects.PickerChangeColor;
// or using PlatformEffects = FormsCommunityToolkit.Effects.UWP.Effects;

var color = (Color)Element.GetValue(RoutingEffects.PickerChangeColor.ColorProperty);


(similar for all the other classes)

File: src/Effects/Effects.iOS/Effects/Entry/EntryDisableAutoCorrect.cs [hunk 9]
```diff
@@ -5,16 +5,15 @@
...
     [Preserve]
-    public class DisableAutoCorrectEffect : PlatformEffect
+    public class EntryDisableAutoCorrect : PlatformEffect
     {
-        UITextSpellCheckingType spellCheckingType;
-        UITextAutocorrectionType autocorrectionType;
-        UITextAutocapitalizationType autocapitalizationType;
+        private UITextSpellCheckingType _spellCheckingType;
+        private UITextAutocorrectionType _autocorrectionType;
+        private UITextAutocapitalizationType _autocapitalizationType;
```
Comment (bartlannoeye) [inline src/Effects/Effects.iOS/Effects/Entry/EntryDisableAutoCorrect.cs]: remove 1 newline

File: src/Effects/Effects.iOS/Effects/Entry/EntryDisableAutoCorrect.cs [hunk 10]
```diff
@@ -23,9 +22,9 @@ protected override void OnAttached()
             if (editText == null) return;
```
Comment (bartlannoeye) [inline src/Effects/Effects.iOS/Effects/Entry/EntryDisableAutoCorrect.cs]: remove 1 newline

File: src/Effects/Effects.iOS/Effects/Switch/SwitchChangeColor.cs [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+[assembly: ExportEffect (typeof (SwitchChangeColor), nameof (SwitchChangeColorEffect))]
+namespace FormsCommunityToolkit.Effects.iOS
+{
+    public class SwitchChangeColor : PlatformEffect
+    {
+        private Color _trueColor;
+        private Color _falseColor;
+
+        public SwitchChangeColor ()
```
Comment (bartlannoeye) [inline src/Effects/Effects.iOS/Effects/Switch/SwitchChangeColor.cs]: unused ctor

File: src/Effects/Effects.iOS/Effects/Switch/SwitchChangeColor.cs [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+        {
+        }
+
+        protected override void OnAttached ()
+        {
+            _trueColor = (Color)Element.GetValue (FormsCommunityToolkit.Effects.SwitchChangeColor.TrueColorProperty);
+            _falseColor = (Color)Element.GetValue (FormsCommunityToolkit.Effects.SwitchChangeColor.FalseColorProperty);
+
+            if (_falseColor != Color.Transparent)
+            {
```
Comment (bartlannoeye) [inline src/Effects/Effects.iOS/Effects/Switch/SwitchChangeColor.cs]: cast to Control once and put in variable

File: src/Effects/Effects.iOS/Effects/Switch/SwitchChangeColor.cs [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+                (Control as UISwitch).TintColor = _falseColor.ToUIColor ();
+                (Control as UISwitch).Layer.CornerRadius = 16;
+                (Control as UISwitch).BackgroundColor = _falseColor.ToUIColor ();
+            }
+
+            if (_trueColor != Color.Transparent)
+                (Control as UISwitch).OnTintColor = _trueColor.ToUIColor ();
+        }
+
+        protected override void OnDetached ()
```
Comment (bartlannoeye) [inline src/Effects/Effects.iOS/Effects/Switch/SwitchChangeColor.cs]: Empty override

File: src/Effects/Effects.Android/Effects/Label/LabelCustomFont.cs (hunk 3 above)
Comment (Depechie) [inline src/Effects/Effects.Android/Effects/Label/LabelCustomFont.cs]: Hmm the class itself has the same name! So full name needed to correctly identify...

File: src/Effects/Effects.UWP/Effects/Entry/EntryRemoveBorder.cs (hunk 6 above)
Comment (Depechie) [inline src/Effects/Effects.UWP/Effects/Entry/EntryRemoveBorder.cs]: Also here, we need the Thickness of UWP not the one from Xamarin forms, so full type name needed

File: src/Effects/Effects.UWP/Effects/Picker/PickerChangeColor.cs (hunk 8 above)
Comment (Depechie) [inline src/Effects/Effects.UWP/Effects/Picker/PickerChangeColor.cs]: It builds / works at my end... maybe best you tweak this to see how result is on your machine?

File: src/Effects/Effects.Android/Effects/Entry/EntryClear.cs (hunk 2 above)
Comment (Depechie) [inline src/Effects/Effects.Android/Effects/Entry/EntryClear.cs]: Need to add the detach because of the inheritance...

**Pull request merged**